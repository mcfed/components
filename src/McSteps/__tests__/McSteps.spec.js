import {shallow} from 'enzyme';
import React from 'react';

import McTestSteps from '../index.tsx';

const setup = props => {
  const wrapper = shallow(<McTestSteps {...props} />);

  return {
    wrapper,
    props
  };
};

describe('快照测试', () => {
  it('全页快照', () => {
    const props = {
      steps: [
        {
          text: '第一步',
          description: '我是第一步',
          path: 'first',
          component: '1'
        },
        {
          text: '第二步',
          description: '我是第二步',
          path: 'second',
          component: '2'
        },
        {
          text: '第三步',
          description: '我是第三步',
          path: 'third',
          component: '3'
        }
      ],
      location: {
        pathname: ''
      }
    };
    const {wrapper} = setup(props);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('全方法测试', () => {
  const props = {
    steps: [
      {
        text: '第一步',
        description: '我是第一步',
        path: 'first',
        component: '1'
      },
      {
        text: '第二步',
        description: '我是第二步',
        path: 'second',
        component: '2'
      },
      {
        text: '第三步',
        description: '我是第三步',
        path: 'third',
        component: '3'
      }
    ],
    location: {
      pathname: '/third'
    },
    history: {
      push: jest.fn()
    }
  };
  it('goRoutes 方法测试', () => {
    const {wrapper} = setup(props);
    const instance = wrapper.instance();
    instance.goRoutes(1);
    expect(instance.state).toEqual({
      currentIndex: 0,
      step: 1
    });
  });
  it('getCurrentStep 方法测试', () => {
    const {wrapper} = setup(props);
    const instance = wrapper.instance();
    expect(instance.getCurrentStep()).toBe(instance.state.step);
  });
});
