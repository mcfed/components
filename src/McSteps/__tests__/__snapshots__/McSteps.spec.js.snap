// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`快照测试 全页快照 1`] = `
<Fragment>
  <Steps
    current={0}
  >
    <Step
      description="我是第一步"
      key="第一步"
      title="第一步"
    />
    <Step
      description="我是第二步"
      key="第二步"
      title="第二步"
    />
    <Step
      description="我是第三步"
      key="第三步"
      title="第三步"
    />
  </Steps>
  <1
    forwardRef={
      Object {
        "current": null,
      }
    }
    getCurrentStep={[Function]}
    goRoutes={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    location={
      Object {
        "pathname": "",
      }
    }
    steps={
      Array [
        Object {
          "component": "1",
          "description": "我是第一步",
          "path": "first",
          "text": "第一步",
        },
        Object {
          "component": "2",
          "description": "我是第二步",
          "path": "second",
          "text": "第二步",
        },
        Object {
          "component": "3",
          "description": "我是第三步",
          "path": "third",
          "text": "第三步",
        },
      ]
    }
  />
</Fragment>
`;
